test/e2e/specs/editor/various/write-design-mode.spec.js

```diff
@@ -187,4 +187,109 @@ test.describe( 'Write/Design mode', () => {
 
 		await expect( nonContentBlock ).toBeHidden();
 	} );
+	test( 'prevents adding blocks to nested containers', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.setContent( '' );
+
+		// Insert a section with a nested block and an editable block.
+		await editor.insertBlock( {
+			name: 'core/group',
+			attributes: {},
+			innerBlocks: [
+				{
+					name: 'core/quote',
+					attributes: {
+						metadata: {
+							name: 'Content block',
+						},
+					},
+					innerBlocks: [
+						{
+							name: 'core/paragraph',
+							attributes: {
+								content: 'Something',
+							},
+						},
+					],
+				},
+			],
+		} );
+
+		// Switch to write mode.
+		await editor.switchEditorTool( 'Write' );
+
+		// Select the inner paragraph block.
+		const paragraph = editor.canvas.getByRole( 'document', {
+			name: 'Block: Paragraph',
+		} );
+		const paragraphClientId = await paragraph.getAttribute( 'data-block' );
+
+		// Select paragraph
+		await paragraph.click();
+
+		// Press enter to check whether it adds a new block.
+		await page.keyboard.press( 'Enter' );
+
+		// Check client id of selected block and see whether it matches paragraphClientId
+		const getSelectedBlock = async () =>
+			await page.evaluate( () =>
+				window.wp.data
+					.select( 'core/block-editor' )
+					.getSelectedBlockClientId()
+			);
+
+		expect( await getSelectedBlock() ).toEqual( paragraphClientId );
```

Comment:
I understand this is from existing tests, but I think it's better to [test user-visible behavior](https://playwright.dev/docs/best-practices#test-user-visible-behavior) when possible. Besides the block editor and us devs one interacts with `clientIds` 😄

```js
await expect( paragraph ).toBeFocused();
// Focused is probably the same as checking the selected class.
// await expect( paragraph ).toHaveClass( /is-selected/ );
await expect( paragraph ).toHaveText( 'Something' );
```

Reply:
It's not a blocker. I also don't mind updating the whole test suite in a follow-up.